samples/bluetooth/iso_broadcast_benchmark/src/broadcaster.c

```diff
@@ -0,0 +1,473 @@
+#include <bluetooth/iso.h>
+#include <sys/byteorder.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(iso_broadcast_broadcaster, LOG_LEVEL_DBG);
+
+#define DEFAULT_BIS_RTN         2
+#define DEFAULT_BIS_INTERVAL_US 7500
+#define DEFAULT_BIS_LATENCY_MS  10
+#define DEFAULT_BIS_PHY         BT_GAP_LE_PHY_2M
+#define DEFAULT_BIS_SDU         CONFIG_BT_ISO_TX_MTU
+#define DEFAULT_BIS_PACKING     0
+#define DEFAULT_BIS_FRAMING     0
+#define DEFAULT_BIS_COUNT       CONFIG_BT_ISO_MAX_CHAN
+
+NET_BUF_POOL_FIXED_DEFINE(bis_tx_pool, CONFIG_BT_ISO_TX_BUF_COUNT,
+			  DEFAULT_BIS_SDU, NULL);
+
+static K_SEM_DEFINE(sem_big_complete, 0, 1);
+static K_SEM_DEFINE(sem_big_term, 0, 1);
+
+static void iso_connected(struct bt_iso_chan *chan)
+{
+	LOG_INF("ISO Channel %p connected", chan);
+	k_sem_give(&sem_big_complete);
+}
+
+static void iso_disconnected(struct bt_iso_chan *chan, uint8_t reason)
+{
+	LOG_INF("ISO Channel %p disconnected with reason 0x%02x",
+	       chan, reason);
+	k_sem_give(&sem_big_term);
+}
+
+static struct bt_iso_chan_ops iso_ops = {
+	.connected	= iso_connected,
+	.disconnected	= iso_disconnected,
+};
+
+static struct bt_iso_chan_io_qos iso_tx_qos = {
+	.interval = DEFAULT_BIS_INTERVAL_US, /* in microseconds */
+	.latency = DEFAULT_BIS_LATENCY_MS, /* milliseconds */
+	.sdu = DEFAULT_BIS_SDU, /* bytes */
+	.rtn = DEFAULT_BIS_RTN,
+	.phy = DEFAULT_BIS_PHY,
+};
+
+static struct bt_iso_chan_qos bis_iso_qos = {
+	.tx = &iso_tx_qos,
+	.packing = DEFAULT_BIS_PACKING, /* 0 - sequential, 1 - interleaved */
+	.framing = DEFAULT_BIS_FRAMING, /* 0 - unframed, 1 - framed */
+};
+
+static struct bt_iso_chan bis_iso_chans[CONFIG_BT_ISO_MAX_CHAN];
+static struct bt_iso_chan *bis[CONFIG_BT_ISO_MAX_CHAN];
+
+static struct bt_iso_big_create_param big_create_param = {
+	.num_bis = DEFAULT_BIS_COUNT,
+	.bis_channels = bis,
+};
+
+static size_t get_chars(char *buffer, size_t max_size)
+{
+	size_t pos = 0;
+
+	while (pos < max_size) {
+		char c = console_getchar();
+
+		if (c == '\n' || c == '\r') {
+			break;
+		}
+		printk("%c", c);
+		buffer[pos++] = c;
+	}
+	printk("\n");
+	buffer[pos] = '\0';
+
+	return pos;
+}
+
+static int parse_rtn_arg(void)
+{
+	char buffer[4];
+	size_t char_count;
+	uint64_t rtn;
+
+	printk("Set RTN (current %u, default %u)\n",
+	       iso_tx_qos.rtn, DEFAULT_BIS_RTN);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+
+	if (char_count == 0) {
+		return DEFAULT_BIS_RTN;
+	}
+
+	rtn = strtoul(buffer, NULL, 0);
+	if (rtn > 16) {
+		printk("Invalid RTN %llu", rtn);
+		return -EINVAL;
+	}
+
+	return (int)rtn;
+}
+
+static int parse_interval_arg(void)
+{
+	char buffer[9];
+	size_t char_count;
+	uint64_t interval;
+
+	printk("Set interval (us) (current %u, default %u)\n",
+	       iso_tx_qos.interval, DEFAULT_BIS_INTERVAL_US);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+
+	if (char_count == 0) {
+		return DEFAULT_BIS_INTERVAL_US;
+	}
+
+	interval = strtoul(buffer, NULL, 0);
+	if (interval < 0x100 || interval > 0xFFFFF) {
```

Comment:
In the spec 8)

There are no such values defined in the host stack at the moment, and in fact we don't even check these values in the host at all before sending it to the controller. I think it would make sense to add them as macros and add checks when creating a BIG, but that would be in a separate PR.